src/test/java/com/code_intelligence/jazzer/mutation/mutator/lang/ByteArrayMutatorTest.java

```diff
@@ -0,0 +1,110 @@
+import com.code_intelligence.jazzer.mutation.annotation.WithLength;
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import com.code_intelligence.jazzer.mutation.support.TestSupport.MockPseudoRandom;
+import com.code_intelligence.jazzer.mutation.support.TypeHolder;
+import org.junit.jupiter.api.Test;
+
+public class ByteArrayMutatorTest {
+  @Test
+  void testBasicFunction() {
+    SerializingMutator<byte @NotNull[]> mutator =
+        (SerializingMutator<byte[]>) LangMutators.newFactory().createOrThrow(
+            new TypeHolder<byte @NotNull[]>() {}.annotatedType());
+    assertThat(mutator.toString()).isEqualTo("byte[]");
+
+    byte[] arr;
+    try (MockPseudoRandom prng = mockPseudoRandom(5, new byte[] {1, 2, 3, 4, 5})) {
+      arr = mutator.init(prng);
+    }
+    assertThat(arr).isEqualTo(new byte[] {1, 2, 3, 4, 5});
+
+    try (MockPseudoRandom prng = mockPseudoRandom()) {
+      arr = mutator.mutate(arr, prng);
+    }
+    assertThat(arr).isEqualTo(new byte[] {2, 4, 6, 8, 10, 6, 7, 8, 9});
+  }
+
+  @Test
+  void testMaxLength() {
+    SerializingMutator<byte[]> mutator =
+        (SerializingMutator<byte[]>) LangMutators.newFactory().createOrThrow(
+            new TypeHolder<byte @NotNull @WithLength(max = 10)[]>() {}.annotatedType());
+    assertThat(mutator.toString()).isEqualTo("byte[]");
+
+    byte[] arr;
+    try (MockPseudoRandom prng = mockPseudoRandom(8, new byte[] {1, 2, 3, 4, 5, 6, 7, 8})) {
+      arr = mutator.init(prng);
+    }
+    assertThat(arr).isEqualTo(new byte[] {1, 2, 3, 4, 5, 6, 7, 8});
+
+    try (MockPseudoRandom prng = mockPseudoRandom()) {
+      arr = mutator.mutate(arr, prng);
+    }
+    assertThat(arr).hasLength(10);
+    assertThat(arr).isEqualTo(new byte[] {2, 4, 6, 8, 10, 12, 14, 16, 9, 10});
+
+    try (MockPseudoRandom prng = mockPseudoRandom()) {
+      arr = mutator.mutate(arr, prng);
+    }
+    assertThat(arr).hasLength(10);
+    assertThat(arr).isEqualTo(new byte[] {3, 6, 9, 12, 15, 18, 21, 24, 18, 20});
+  }
+
+  @Test
+  void testMaxLengthInitClamp() {
+    SerializingMutator<byte[]> mutator =
+        (SerializingMutator<byte[]>) LangMutators.newFactory().createOrThrow(
+            new TypeHolder<byte @NotNull @WithLength(max = 5)[]>() {}.annotatedType());
+    assertThat(mutator.toString()).isEqualTo("byte[]");
+
+    byte[] arr;
+    try (MockPseudoRandom prng = mockPseudoRandom(10, new byte[] {1, 2, 3, 4, 5, 6, 7, 8, 9, 10})) {
+      arr = mutator.init(prng);
+    } catch (AssertionError e) {
+      assertThat(e).isNotNull();
+    }
+  }
+
+  @Test
+  void testMinLengthInitClamp() {
+    SerializingMutator<byte[]> mutator =
+        (SerializingMutator<byte[]>) LangMutators.newFactory().createOrThrow(
+            new TypeHolder<byte @NotNull @WithLength(min = 5)[]>() {}.annotatedType());
+    assertThat(mutator.toString()).isEqualTo("byte[]");
+
+    // The mock libfuzzer mutator doesn't currently support outputting less data than it's given, so
```

Comment:
Feel free to modify the mock mutator so that you can test this case more easily. I just made up its implementation on the spot, no reason to feel attached to it. :-)

Reply:
I spent a bit of time thinking about how I could change it but wasn't sure what a better method could be since there's no prng to hook into. Maybe use some of the elements in the array as instructions to modify the rest of it? Like have the first element be the new array's length?

Reply:
Yes, that would be an option. If it is really only for this one test, you could also have the mock mutator get the length from e.g. `System.getProperty`. For test mocks I wouldn't consider that too hacky.

Reply:
I went with `System.getProperty` as that seemed the easiest to understand though I think it does mean you can't run these tests in parallel but I don't think that's a very common usecase anyway.